[Variant analysis] New tutorial - "Identification of allelic variants in SARS-CoV-2 from deep sequencing reads" (#2689)
galaxyproject/training-material: topics/variant-analysis/tutorials/sars-cov-2-variant-discovery/tutorial.md

To upload the data, there are several possibilities depending of how many datasets and what their origin is:

- Import datasets (locally, on a given URL or from a shared data library) and organize them as a dataset colletion

  A dataset collection is a way to represent an arbitrarily large collection of samples as a singular entity within a user's workspace.

- Import from [NCBI's Sequence Read Archive (SRA) at NCBI](https://www.ncbi.nlm.nih.gov/sra)

   > ### {% icon comment %} Getting data from SRA
   >
   > [A dedicated tutorial is available to explain how to find and import SARS-CoV-2 data from SRA]({% link topics/variant-analysis/tutorials/sars-cov-2/tutorial.md %}).
   >
   {: .comment} 

> ### {% icon hands_on %} Hands-on: Import datasets
>
> 1. Import the datasets
>
>    - Option 1: From your own local data using **Upload Data** (1-10 datasets)
>      
>      {% snippet faqs/galaxy/datasets_upload.md %}
>
>    - Option 2: From your own local data using **FTP** (>10 datasets)
>
>      {% snippet faqs/galaxy/datasets_upload_ftp.md %}
>
>    - Option 3: From the shared data library
>
>      {% snippet faqs/galaxy/datasets_import_from_data_library.md %}
>

> >      For our example datasets, the 28 `fastqsanger.gz` files in folder `GTN - Material` -> `Variant analysis` -> `Identification of allelic variants in SARS-CoV-2 from deep sequencing reads`

Reviewer: ```suggestion
>      For our example datasets, the 36 `fastqsanger.gz` files (representing 18 samples) in folder `GTN - Material` -> `Variant analysis` -> `Identification of allelic variants in SARS-CoV-2 from deep sequencing reads`
```

>
>    - *"2: DP Filer*: `1`
>
>       The minimum depth of all alignment at a variant site
>
>    - *"3: Variation data to report"*: `Final (SnpEff-) annotated variants`
>
>       The collection with variation data in VCF format: the output of the previous workflow
>
>    - *"4: gene products translations"*: `NC_045512.2_feature_mapping.tsv` or `NC_045512.2 feature mapping`
>
>       The custom tabular file mapping NCBI RefSeq Protein identifiers (as used by snpEff version 4.5covid19) to their commonly used names, part of the auxillary data.
>
{: .hands_on}

Both workflows generate several outputs. Most of them are collections with results for each samples. There are also reports combining information for all samples.

1. **Combined Variant Report by Sample**: table sumarize several information (columns) for each AVs in each sample (row)

   Column | Field | Meaning
   --- | --- | ---
   1 | `Sample` | SRA run ID
   2 | `POS` | Position in [NC_045512.2](https://www.ncbi.nlm.nih.gov/nuccore/1798174254)
   3 | `FILTER` | `Filter` field from VCF
   4 | `REF` |  Reference base
   5 | `ALT` | Alternative base
   6 | `DP` | Sequencing depth
   7 | `AF` | Alternative allele frequency

>    8 | `SB` | Alternative allele frequency

Reviewer: Remove this duplicated line

Author: Fixed

35 | `qc.privateMutations.status` | Status for ambiguous nucleotides
36 | `qc.privateMutations.total` | Number of ambiguous nucleotides
37 | `qc.snpClusters.clusteredSNPs` | Clusters with 6 or more differences in 100 bases
38 | `qc.snpClusters.score` | Score for clustered differences
39 | `qc.snpClusters.status` | Status for clustered differences
40 | `qc.snpClusters.totalSNPs` | Number of differences in clusters
41 | `errors` | Other errors (e.g. sequences in which some of the major genes fail to translate because of frame shifting insertions or deletions)

> ### {% icon question %} Questions
> 
> How many different lineages have been found? How many samples for each lineage?
>
> ![Illustration of phylogenetic relationship of clades, as used in Nextclade](../../images/sars-cov-2-variant-discovery/ncov_clades.png "Illustration of phylogenetic relationship of clades, as used in Nextclade (Source: <a href="https://clades.nextstrain.org/">Nextclade</a>)")
>
> > ### {% icon solution %} Solution
> >
> > To summarize the number of lineages and number of samples for  each lineage, we can run {% tool [Group data](Grouping1) %} with the following parameters:

> > >    - {% icon param-file %} *"Select data"*: output of **pangolin**

Reviewer: ```suggestion
> >    - {% icon param-file %} *"Select data"*: output of **Nextclade**
```

---
layout: tutorial_hands_on

title: Identification of allelic variants in SARS-CoV-2 from deep sequencing reads

> level: Introductory

Reviewer 2: is that true?

Reviewer 3: I would say Intermediate

zenodo_link: "https://zenodo.org/record/5036687"
questions:
- How can we extract annotated allelic variants in SARS-Cov-2 sequences in Galaxy?
- Which tools can we use to identify SARS-CoV-2 clades in Galaxy?
objectives:
- Repeat SARS-CoV-2 data preparation
- Select and execute workflow to extract annotated allelic variants from FASTQ files

> - Execute workflow to summarize annotated allelic variants

Reviewer 2: Run workflow to summarize and generat report for previously called allelic variants

- Interpret summaries for annotated allelic variants

> - Execute workflow to extract consensus sequences

Reviewer 2: Run instead of execut?

- Execute tools to assign clades/lineages
time_estimation: 3H
key_points:

> - 4 workflows are possible to extract annotated allelic variants from FASTQ files depending on the type of input

Reviewer 2: 4 specilised variant calling workflows for the identification of annotated allelic variants from RAW data

- Many datasets can be processed in parallel using collections
- Annotated allelic variants can be used to identify SARS-CoV-2 clades/lineages in each samples using consensus sequences
contributors:
- wm75
- bebatut
tags:
- covid19
---

# Introduction
{:.no_toc}

Effectively monitoring global infectious disease crises, such as the COVID-19 pandemic, requires capacity to generate and analyze large volumes of sequencing data in near real time. These data have proven essential for monitoring the emergence and spread of new variants, and for understanding the evolutionary dynamics of the virus.

Two sequencing platforms in combination with several established library preparation strategies are predominantly used to generate SARS-CoV-2 sequence data. However, data alone do not equal knowledge: they need to be analyzed. The Galaxy community developed analysis workflows to support the **identification of allelic variants (AVs) in SARS-CoV-2 from deep sequencing reads**. 

> These workflows allow one to identify AVs and lineages in SARS-CoV-2 genomes with variant allele frequencies ranging from 5% to 100% (i.e., they detect variants with intermediate frequencies as well.

Reviewer 2: What are AV? long version here?

Reviewer 2: Can we add a link to covid19.galaxyproject.org here ... more information, more benchmarks etc..

- Single end data derived from Illumina-based RNAseq experiments
- Paired end data derived from Illumina-based RNAseq experiments
- Paired-end data generated with Illumina-based Ampliconic (ARTIC) protocols
- ONT fastq files generated with Oxford nanopore (ONT)-based Ampliconic (ARTIC) protocols

> ### Agenda
>
> In this tutorial, we will cover:
>
> 1. TOC
> {:toc}
>
{: .agenda}

# Prepare Galaxy and data

Any analysis should get their own Galaxy history. So let's start by creating a new one:

> ### {% icon hands_on %} Hands-on: Prepare the Galaxy history
>
> 1. Create a new history for this analysis
>
>    {% snippet faqs/galaxy/histories_create_new.md %}
>
> 2. Rename the history
>
>    {% snippet faqs/galaxy/histories_rename.md %}
>
{: .hands_on}

## Import auxiliary datasets

For extracting the variants, we need to get the SARS-CoV-2 reference (a sequence-identical copy of [NC_045512.2](https://www.ncbi.nlm.nih.gov/nuccore/NC_045512.2?report=fasta)) and gene name aliases (to give gene regions easily recognizable names), i.e. a custom tabular file mapping NCBI RefSeq Protein identifiers as used by snpEff version 4.5covid19 to their commonly used names. 

> In addition, we need for datasets generated using ARTIC protocols 2 extra datasets: a BED file with ARTIC network primer scheme and a custom tabular file describing the amplicon grouping of the primers in the primer scheme file.

Reviewer 2: "we need for datasets" sounds strange

>      {{ page.zenodo_link }}/ERR5949462_2.fastqsanger.gz
>      {{ page.zenodo_link }}/ERR5949463_1.fastqsanger.gz
>      {{ page.zenodo_link }}/ERR5949463_2.fastqsanger.gz
>      {{ page.zenodo_link }}/ERR5949464_1.fastqsanger.gz
>      {{ page.zenodo_link }}/ERR5949464_2.fastqsanger.gz
>      {{ page.zenodo_link }}/ERR5949465_1.fastqsanger.gz
>      {{ page.zenodo_link }}/ERR5949465_2.fastqsanger.gz
>      {{ page.zenodo_link }}/ERR5949466_1.fastqsanger.gz
>      {{ page.zenodo_link }}/ERR5949466_2.fastqsanger.gz
>      {{ page.zenodo_link }}/ERR5949467_1.fastqsanger.gz
>      {{ page.zenodo_link }}/ERR5949467_2.fastqsanger.gz
>      {{ page.zenodo_link }}/ERR5949468_1.fastqsanger.gz
>      {{ page.zenodo_link }}/ERR5949468_2.fastqsanger.gz
>      {{ page.zenodo_link }}/ERR5949469_1.fastqsanger.gz
>      {{ page.zenodo_link }}/ERR5949469_2.fastqsanger.gz
>      ```
>
> 2. Create a collection to organize the data
>
>    - Option 1: Single-end data (Illumina or ONT data)
>
>      {% snippet faqs/galaxy/collections_build_list.md %}
>
>    - Option 2: Paired-end data (Illumina data)
>
>      {% snippet faqs/galaxy/collections_build_list_paired.md %}
>

> >      *To do for example datasets: the example datasets are paired-end data with datasets with `_1` contain the forward reads and datasets with `_2` the reverse reads.*

Reviewer 2: To do for example?

>        Illumina RNAseq PE | [{% icon workflow %}](https://usegalaxy.org/workflows/run?id=5a610d5a42d50cf3) | [{% icon workflow %}](https://usegalaxy.eu/workflows/run?id=e7ba6ca41d46baf2) | [{% icon workflow %}](https://usegalaxy.org.au/workflows/run?id=bd1cf22d47389742)
>        Illumina RNAseq SE | [{% icon workflow %}](https://usegalaxy.org/workflows/run?id=c092a3631d68ce38) | [{% icon workflow %}](https://usegalaxy.eu/workflows/run?id=6b41f7afbe14647d) | [{% icon workflow %}](https://usegalaxy.org.au/workflows/run?id=31dbd313e5c8160b)
>        ONT ARTIC | [{% icon workflow %}](https://usegalaxy.org/workflows/run?id=88d0b64011c3148c) | [{% icon workflow %}](https://usegalaxy.eu/workflows/run?id=17d7f9bddbc834f8) | [{% icon workflow %}](https://usegalaxy.org.au/workflows/run?id=0c6f7bfc826433c4)
>
>      The browser will open a new tab with Galaxy's workflow invocation interface.
>
>    - Option 2: Import the workflow
>

> >      - Copy the URL (e.g. via right-click) of the appropriate workflow version or download it to your computer:

Reviewer 2: Should we just include the workflowhub import? or use the IWC URLs?

Author: Updated

>      - {% icon param-collection %} *"3: Collection of ONT-sequenced reads"*: dataset collection created for the input datasets
{: .hands_on}

The execution of the workflow takes some time. It is possible to launch the next step even if it is not done, as long as all steps are successfully scheduled.

# From annotated AVs per sample to AV summary

Once the jobs of previous workflow are done, we identified AVs for each sample. We can run a "Reporting workflow" on them to generate a final AV summary. It  takes a table of AVs produced by any of the other four workflows and generates a list of AVs by Samples and by Variant. 

> ### {% icon hands_on %} Hands-on: From annotated AVs per sample to AV summary
>
> 1. **Get the workflow** on Galaxy
>    
>    - Option 1: Use workflow available on
>      - [usegalaxy.org](https://usegalaxy.org/workflows/run?id=2967ef82911f2cca)
>      - [usegalaxy.eu](https://usegalaxy.eu/workflows/run?id=0d10c137a0f08bca)
>      - [usegalaxy.org.au](https://usegalaxy.org.au/workflows/run?id=8e2b94c6fbf44368)
>
>    - Option 2: Import the workflow
>

> >      - Copy the URL (e.g. via right-click) of [the workflow](https://raw.githubusercontent.com/galaxyproject/iwc/main/workflows/sars-cov-2-variant-calling/sars-cov-2-variation-reporting/variation-reporting.ga) or download it to your computer        

Reviewer 2: We could import those from Workflowhub

Author: Updated

>
> More details about the workflows, including benchmarking of the tools, can be found on [covid19.galaxyproject.org](https://covid19.galaxyproject.org/genomics/global_platform/#methods)
{: .details}

> ### {% icon hands_on %} Hands-on: From FASTQ to annotated AVs
>

> > 1. **Get the workflow** on Galaxy 

Reviewer: Include a third option - import from iwc-workflows github organization via link - for people working on instances without TRS configured?

Author: Added

>
>    - Option 1: Use workflows directly on [usegalaxy.eu](https://usegalaxy.eu/) from [WorkflowHub](https://workflowhub.eu)
>      - Open the workflow page on WokflowHub
>        - [Illumina ARTIC PE](https://workflowhub.eu/workflows/110) - The one to use for example datasets
>        - [Illumina RNAseq SE](https://workflowhub.eu/workflows/112)
>        - [Illumina RNAseq PE](https://workflowhub.eu/workflows/113)
>        - [ONT ARTIC](https://workflowhub.eu/workflows/111)
>      - Click on `Run on usegalaxy.eu` on the top right of the page
>      
>        The browser will open a new tab with Galaxy's workflow invocation interface.
>

> >    - Option 2: Import the workflows of IWC (Intergalactic Workflow Commission) from WorkflowHub using the workflow search

Reviewer: Maybe provide instructions here for import from dockstore cause that will work on usegalaxy.org as well.

Reviewer: This has the additional advantage that the sixth (the consensus) WF is available there.

>

> >      {% snippet faqs/galaxy/workflows_import_search.md trs_server="workflowhub.eu" search_query='organization:"IWC Workflows"' %}

Reviewer: ```suggestion
>      {% snippet faqs/galaxy/workflows_import_search.md trs_server="Dockstore" search_query='organization:"iwc-workflows"' %}
```

> - The workflow for ONT-sequenced ARTIC data (ONT ARTIC) is modeled after the alignment/AV-calling steps of the [ARTIC pipeline](https://artic.readthedocs.io/). It performs, essentially, the same steps as that pipeline’s minion command, i.e. read mapping with **minimap2** and AV calling with **medaka**. Like the Illumina ARTIC workflow it uses **ivar** for primer trimming. Since ONT-sequenced reads have a much higher error rate than Illumina-sequenced reads and are therefore plagued more by false-positive AV calls, this workflow makes no attempt to handle amplicons affected by potential primer-binding site mutations.
>
> All four workflows use **SnpEff**, specifically its 4.5covid19 version, for AV annotation.
>

> > Workflows default to requiring an AF ≥ 0.05 and AV-supporting reads of ≥ 10 (these and all other parameters can be easily changed by the user). For an AV to be listed in the reports it must surpass these thresholds in at least one sample of the respective dataset. We estimate that for AV calls with an AF ≥ 0.05 our analyses have a false-positive rate of < 15% for both Illumina RNAseq and Illumina ARTIC data, while the true-positive rate of calling such low-frequency AVs is ~80% and approaches 100% for AVs with an AF ≥ 0.15. This estimate is based on an initial application of the Illumina RNAseq and Illumina ARTIC workflows to two samples for which data of both types had been obtained at the virology department of the University of Freiburg and the assumption that AVs supported by both sets of sequencing data are true AVs. The second threshold of 10 AV-supporting reads is applied to ensure that calculated AFs are sufficiently precise for all AVs.

Reviewer 3: ```suggestion
> Workflows default to requiring an AF ≥ 0.05 and AV-supporting reads of ≥ 10 (these and all other parameters can be easily changed by the user). For an AV to be listed in the reports, it must surpass these thresholds in at least one sample of the respective dataset. We estimate that for AV calls with an AF ≥ 0.05, our analyses have a false-positive rate of < 15% for both Illumina RNAseq and Illumina ARTIC data, while the true-positive rate of calling such low-frequency AVs is ~80% and approaches 100% for AVs with an AF ≥ 0.15. This estimate is based on an initial application of the Illumina RNAseq and Illumina ARTIC workflows to two samples for which data of both types had been obtained at the virology department of the University of Freiburg and the assumption that AVs supported by both sets of sequencing data are true AVs. The second threshold of 10 AV-supporting reads is applied to ensure that calculated AFs are sufficiently precise for all AVs.
```

Author: I copied it from the covid19 Galaxy website :sweat_smile:

>
> More details about the workflows, including benchmarking of the tools, can be found on [covid19.galaxyproject.org](https://covid19.galaxyproject.org/genomics/global_platform/#methods)
{: .details}

> ### {% icon hands_on %} Hands-on: From FASTQ to annotated AVs
>
> 1. **Get the workflow** on Galaxy 
>
>    - Option 1: Use workflows directly on [usegalaxy.eu](https://usegalaxy.eu/) from [WorkflowHub](https://workflowhub.eu)
>

> >      - Open the workflow page on WokflowHub

Reviewer 3: ```suggestion
>      - Open the workflow page on the WorkflowHub
```

>      - {% icon param-file %} *"2: NC_045512.2 FASTA sequence of SARS-CoV-2"*: `NC_045512.2_reference.fasta` or `NC_045512.2 reference sequence`
>
>    - For **Illumina RNAseq PE** workflow (named **COVID-19: variation analysis on WGS SE data**)
>      - {% icon param-collection %} *"1: Input dataset collection"*: dataset collection created for the input datasets
>      - {% icon param-file %} *"2: NC_045512.2 FASTA sequence of SARS-CoV-2"*: `NC_045512.2_reference.fasta` or `NC_045512.2 reference sequence`
>
>    - For **ONT ARTIC** workflow (named **COVID-19: variation analysis of ARTIC ONT data**)
>      - {% icon param-file %} *"1: ARTIC primer BED"*: `ARTIC_nCoV-2019_v3.bed` or `ARTIC nCoV-2019 v3`
>      - {% icon param-file %} *"2: FASTA sequence of SARS-CoV-2"*: `NC_045512.2_reference.fasta` or `NC_045512.2 reference sequence`
>      - {% icon param-collection %} *"3: Collection of ONT-sequenced reads"*: dataset collection created for the input datasets
{: .hands_on}

The execution of the workflow takes some time. It is possible to launch the next step even if it is not done, as long as all steps are successfully scheduled.

# From annotated AVs per sample to AV summary

> Once the jobs of previous workflow are done, we identified AVs for each sample. We can run a "Reporting workflow" on them to generate a final AV summary. It  takes a table of AVs produced by any of the other four workflows and generates a list of AVs by Samples and by Variant. 

Reviewer 3: ```suggestion
Once the jobs of previous workflows are done, we identified AVs for each sample. We can run a "Reporting workflow" on them to generate a final AV summary. It takes a table of AVs produced by any of the other four workflows and generates a list of AVs by Samples and by Variant. 
```

   Column | Field | Meaning
   --- | --- | ---
   1 | `POS` | Position in [NC_045512.2](https://www.ncbi.nlm.nih.gov/nuccore/1798174254)
   2 | `REF` | Reference base
   3 | `ALT` | Alternative base
   4 | `IMPACT` | Functional impact (from SNPEff) 
   5 | `FUNCLASS` | Funclass for change (from SNPEff)
   6 | `EFFECT` | Effect of change (from SNPEff)
   7 | `GENE` | Gene 
   8 | `CODON` | Codon 
   9 | `AA` | Amino acid 
   10 |`TRID` | Short name for the gene 
   11 |`countunique(Sample)` | Number of distinct samples containing this change 
   12 |`min(AF)` | Minimum Alternative Allele Freq across all samples containing this change 
   13 |`max(AF)` | Maximum Alternative Allele Freq across all samples containing this change 
   14 |`SAMPLES(above-thresholds)` | List of distinct samples where this change has frequency abobe threshold (5%)
   15 |`SAMPLES(all)` | List of distinct samples containing this change at any frequency (including below threshold) 
   16 |`AFs(all)` | List of all allele frequencies across all samples 
   17 |`change` |  Change 

   > ### {% icon question %} Questions
   > 
   > 1. How many AVs are found?

>    > 1. What are the different impacts of the AVs ?

Reviewer 3: ```suggestion
   > 1. What are the different impacts of the AVs?
```

Reviewer 3: not sure why 1 is twice

   > 2. How many variants are found for each impact?
   > 3. What are the different effects for HIGH impact?
   > 4. Is there any AVs impacting all samples?
   >
   > > ### {% icon solution %} Solution
   > >
   > > 1. By expanding the dataset in the history, we have the number of lines in the file. 191 lines for the example datasets. The first line is the header of the table. Then 190 AVs.
   > >
   > > 2. The different impacts of the AVs are HIGH, MODERATE and LOW.
   > >

>    > > 2. To get the number of AVs for each impact levels, we can run {% tool [Group data](Grouping1) %} with the following parameters:

Reviewer 3: and here 2 is twice

Author: It should not matter. Markdown resolves it :smile:

   > >
   > >    For our example datasets, 4 AVs are found for all samples
   > {: .solution}
   {: .question}

3. **Variant frequency**

   ![Variant frequency plot](../../images/sars-cov-2-variant-discovery/variant-frequency.svg)

   This plot represents AFs (cell color) for the different AVs (columns) and the different samples (rows). The AVs are grouped by genes (different colors on the 1st row). Information about their effect is also represented on the 2nd row. The samples are clustered following the tree displayed on the left.

   In the example datasets, the samples are clustered in 3 clusters (as we defined when running the workflow), that may represent different SARS-CoV-2 variants as the AVs profiles are different.

# From AVs to consensus sequences

For the variant calls, we can now run a workflow which generates reliable consensus sequences according to transparent criteria that capture at least some of the complexity of variant calling:

- Each consensus sequence is guaranteed to capture all called, filter-passing variants as defined in the VCF of its sample that reach a user-defined consensus allele frequency threshold.

> - Filter-failing variants and variants below a second user-defined minimal allele frequency threshold are be ignored.

Reviewer 3: ```suggestion
- Filter-failing variants and variants below a second user-defined minimal allele frequency threshold are ignored.
```

>
>       Only variant calls with an AF greater than this value will be considered consensus variants.
>
>    - *"3: min-AF for failed variants"*: `0.25`
>
>       Variant calls with an AF higher than this value, but lower than the AF threshold for consensus variants will be considered questionable and the respective sites be masked (with Ns) in the consensus sequence.
>
>    - *"4: aligned reads data for depth calculation"*: `Full processed reads for variant calling`
>
>       Collection with fully processed BAMs generated by the first workflow. 
>
>       For ARTIC data, the BAMs should NOT have undergone processing with **ivar removereads**
>
>    - *"5: Depth-threshold for masking"*: `5`
>
>       Sites in the viral genome covered by less than this number of reads are considered questionable and will be masked (with Ns) in the consensus sequence independent of whether a variant has been called at them or not.
>
>    - *"6: Reference genome*": `NC_045512.2_reference.fasta` or `NC_045512.2 reference sequence`
>
>       SARS-CoV-2 reference genome, as part of the auxillary data.
>
{: .hands_on}

The main outputs of the workflow are:

> - a collection of viral consensus sequences

Reviewer 3: ```suggestion
- A collection of viral consensus sequences.
```

> - a multisample FASTA of all these sequences.

Reviewer 3: ```suggestion
- A multisample FASTA of all these sequences.
```

The last one can be given as input for tools like **Pangolin** or **Nextclade**.

# From consensus sequences to clade assignations

To assign lineages to the different samples from their consensus sequences, 2 tools are available **Pangolin** or **Nextclade**

## With Pangolin

Pangolin (Phylogenetic Assignment of Named Global Outbreak LINeages) can be used to assign a SARS-CoV-2 genome sequence the most likely lineage based on the PANGO nomenclature system.

> ### {% icon hands_on %} Hands-on: From consensus sequences to clade assignations using Pangolin
>
> 1. {% tool [Pangolin](toolshed.g2.bx.psu.edu/repos/iuc/pangolin/pangolin/3.1.4+galaxy0) %} with the following parameters:
>    - {% icon param-file %} *"Input FASTA File(s)"*: `Multisample consensus FASTA`
>
> 2. Inspect the generated output
{: .hands_on}

Pangolin generates a table file with taxon name and lineage assigned. Each line corresponds to each sample in the input consensus FASTA file provided. The columns are:

Column | Field | Meaning
--- | --- | ---

> 1 | `taxon` | The name of an input query sequence, here the sample name

Reviewer 3: ```suggestion
1 | `taxon` | The name of an input query sequence, here the sample name.
```

4 | `ambiguity_score` | This score is a function of the quantity of missing data in a sequence. It represents the proportion of relevant sites in a sequnece which were imputed to the reference values. A score of 1 indicates that no sites were imputed, while a score of 0 indicates that more sites were imputed than were not imputed. This score only includes sites which are used by the decision tree to classify a sequence.
5 | `scorpio_call` | If a query is assigned a constellation by scorpio this call is output in this column. The full set of constellations searched by default can be found at the constellations repository.
6 | `scorpio_support` | The support score is the proportion of defining variants which have the alternative allele in the sequence.

> 7 | `scorpio_conflict` | The conflict score is the proportion of defining variants which have the reference allele in the sequence. Ambiguous/other non-ref/alt bases at each of the variant positions contribute only to the denominators of these scores

Reviewer 3: ```suggestion
7 | `scorpio_conflict` | The conflict score is the proportion of defining variants which have the reference allele in the sequence. Ambiguous/other non-ref/alt bases at each of the variant positions contribute only to the denominators of these scores.
```

> 8 | `version` | A version number that represents both the pango-designation number and the inference engine used to assign the lineage

Reviewer 3: ```suggestion
8 | `version` | A version number that represents both the pango-designation number and the inference engine used to assign the lineage.
```

9 | `pangolin_version` | The version of pangolin software running.

> 10 | `pangoLEARN_version` | The dated version of the pangoLEARN model installed

Reviewer 3: ```suggestion
10 | `pangoLEARN_version` | The dated version of the pangoLEARN model installed.
```

> 11 | `pango_version` | The version of pango-designation lineages that this assignment is based on

Reviewer 3: ```suggestion
11 | `pango_version` | The version of pango-designation lineages that this assignment is based on.
```

> 12 | `status` | Indicates whether the sequence passed the QC thresholds for minimum length and maximum N content

Reviewer 3: ```suggestion
12 | `status` | Indicates whether the sequence passed the QC thresholds for minimum length and maximum N content.
```

> 13 | `note` | If any conflicts from the decision tree, this field will output the alternative assignments. If the sequence failed QC this field will describe why. If the sequence met the SNP thresholds for scorpio to call a constellation, it’ll describe the exact SNP counts of Alt, Ref and Amb (Alternative, reference and ambiguous) alleles for that call

Reviewer 3: ```suggestion
13 | `note` | If any conflicts from the decision tree, this field will output the alternative assignments. If the sequence failed QC this field will describe why. If the sequence met the SNP thresholds for scorpio to call a constellation, it’ll describe the exact SNP counts of Alt, Ref and Amb (Alternative, Reference and Ambiguous) alleles for that call.
```

> ### {% icon hands_on %} Hands-on: From consensus sequences to clade assignations using Nextclade
>
> 1. {% tool [Nextclade](toolshed.g2.bx.psu.edu/repos/iuc/nextclade/nextclade/0.14.4+galaxy0) %} with the following parameters:
>    - {% icon param-file %} *"SARS-CoV-2 consensus sequences (FASTA)"*: `Multisample consensus FASTA`
>    - {% icon param-check %} *"Output options"*: `Tabular format report`
>
> 2. Inspect the generated output
{: .hands_on}

Column | Field | Meaning
--- | --- | ---
1 | `seqName` | Name of the sequence in the source data, here the sample name
2 | `clade` | The result of the clade assignment of a sequence, as defined by Nextstrain. Currently known clades are depicted in the schema below
3 | `qc.overallScore` | Overall QC score
4 | `qc.overallStatus` | Overall QC status
5 | `totalGaps` | Number of - characters (gaps)
6 | `totalInsertions` | Total length of insertions
7 | `totalMissing` | Number of N characters (missing data) 
8 | `totalMutations` | Number of mutations. Mutations are called relative to the reference sequence Wuhan-Hu-1
9 | `totalNonACGTNs` | Number of non-ACGTN characters

> 10 | `totalPcrPrimerChanges` |

Author: I could not find a good explanation for this field. @wm75 do you know the meaning of this field?

> 10 | `totalPcrPrimerChanges` |

Reviewer: https://github.com/nextstrain/nextclade/blob/master/docs/user/nextclade-algorithm.md#5-pcr-primer-changes-detection

Never really looked at this feature before and turns out, you cannot currently pass in the primer info file to the Galaxy tool cause Advanced options -> Custom PCR primer sites is restricted to JSON when nextclade actually expects CSV.

> 10 | `totalPcrPrimerChanges` |

Reviewer: Maybe you can write sth like: Number of mutations affecting user-specified PCR primer binding sites (not used in this example).

11 | `substitutions` | List of mutations
12 | `deletions` | List of deletions (positions are 1-based)
13 | `insertions` | Insertions relative to the reference Wuhan-Hu-1 (positions are 1-based)
14 | `missing` | Intervals consisting of N characters
15 | `nonACGTNs` | List of positions of non-ACGTN characters (for example ambiguous nucleotide codes)

> 16 | `pcrPrimerChanges` |

Author: Same as above :sweat_smile:
